### PR TITLE
Add N-API documentation

### PR DESCRIPTION
**Pull request by DanielBallaSZTE**
IoT.js-DCO-1.0-Signed-off-by: Daniel Balla dballa@inf.u-szeged.hu

### CODE REVIEW

File: docs/api/IoT.js-API-N-API.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+# N-API
+
+N-API (pronounced N as in the letter, followed by API) is an API for building native Addons. N-API is generally used to create and manipulate JavaScript values. You can read more about N-API [here](https://nodejs.org/api/n-api.html#n_api_n_api).
+
+### Building N-API with IoT.js
+
+You need to append the `--n-api` parameter to the build command (e.g. `tools/build.py --n-api`).
+This automatically adds the N-API module to IoT.js.
+
+To run N-API test you should also append `-n--api` after the `testrunner.py`.
```
Comment (LaszloLango) [inline docs/api/IoT.js-API-N-API.md]: typo: `--n-api`

File: docs/api/IoT.js-API-N-API.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+| [napi_typeof](https://nodejs.org/api/n-api.html#n_api_napi_typeof)  | O |
+| [napi_unwrap](https://nodejs.org/api/n-api.html#n_api_napi_unwrap)  | O |
+| [napi_value](https://nodejs.org/api/n-api.html#n_api_napi_value)  | O |
+| [napi_valuetype](https://nodejs.org/api/n-api.html#n_api_napi_valuetype)  | O |
+| [napi_wrap](https://nodejs.org/api/n-api.html#n_api_napi_wrap)  | O |
+
+
+# N-API
+
+N-API (pronounced N as in the letter, followed by API) is an API for building native Addons. N-API is generally used to create and manipulate JavaScript values. You can read more about N-API [here](https://nodejs.org/api/n-api.html#n_api_n_api).
```
Comment (LaszloLango) [inline docs/api/IoT.js-API-N-API.md]: The link should point to the LTS version not to the latest.

Comment (DanielBallaSZTE) [reply docs/api/IoT.js-API-N-API.md]: Right, changed all of the links to the LTS version.

**Comment by DanielBallaSZTE** [comment]
@LaszloLango I've updated the PR

**Pull request merged**